### PR TITLE
fuchsia: Handle multiple views in platformViews path

### PR DESCRIPTION
**Pull request by arbreng**
The new platformViews path we are introducing for Fuchsia is not able to handle multiple platform views at once.  This PR fixes that and enables proper view IDs with platform_view channel events.

Fixes: https://bugs.fuchsia.dev/p/fuchsia/issues/detail?id=71337

### CODE REVIEW

**Comment by freiling** [comment]
I'm on PTO through monday, I can look at this on Tuesday if you want but probably not before that

File: flow/scene_update_context.cc
Comment (jaeheon) [inline flow/scene_update_context.cc:80]: I'm used to seeing at least a DCHECK() for a raw pointer passed in.

File: flow/scene_update_context.h
Comment (jaeheon) [inline flow/scene_update_context.h:151]: = nullptr; 
just in case a ctor forgets.

File: shell/platform/fuchsia/BUILD.gn
Comment (jaeheon) [inline shell/platform/fuchsia/BUILD.gn:90]: disabled deliberately?

File: shell/platform/fuchsia/flutter/engine.cc
Comment (jaeheon) [inline shell/platform/fuchsia/flutter/engine.cc:187]: Just curious if there's a thread_id facility to DCHECK in the error closure.

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc
Comment (jaeheon) [inline shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc:526]: scenic_views_.count(view_id) == 0
is a slightly clearer way of saying it too

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc
Comment (jaeheon) [inline shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc:548]: Just curious if
scenic_views_[view_id] = std::move(new_view);
would work?

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc
Comment (jaeheon) [inline shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc:559]: Is a call to SetViewProperties ensured to never follow a call to DestroyView? Is it guaranteed because it's tied to the Dart object lifecycle?

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.h
Comment (jaeheon) [inline shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.h:51]: Do these match the default FIDL values?

File: shell/platform/fuchsia/flutter/platform_view.h
Comment (jaeheon) [inline shell/platform/fuchsia/flutter/platform_view.h:40]: Probably a little table would help make the connections clear:

// raster thread - FuchsiaExternalViewEmbedder - fuchsia.ui.scenic.Session client - send commands
// platform thread - flutter::PlatformView  - fuchsia.ui.scenic.SessionListener server - receive events

**Review by jaeheon — COMMENTED** [review]
First time reviewing code here... :)

File: flow/view_holder.cc
Comment (jaeheon) [inline flow/view_holder.cc:80]: It looks like the ViewHolder object is left in bindings. Is the object itself destroyed elsewhere?

File: flow/view_holder.cc
Comment (jaeheon) [inline flow/view_holder.cc:123]: I see pending_view_holder_token_ was moved out earlier (L 114) to the ViewHolder ctor. Wouldn't this always be an empty value here?

**Review by jaeheon — COMMENTED** [review]
(need to break for lunch, will continue after)

File: flow/scene_update_context.cc
Comment (arbreng) [reply flow/scene_update_context.cc:80]: Will add DCHECK

File: shell/platform/fuchsia/flutter/platform_view.h
Comment (jaeheon) [inline shell/platform/fuchsia/flutter/platform_view.h:186]: Would it be just cleaner to say
std::map<ResourceId, zx_handle_t>
?

File: shell/platform/fuchsia/flutter/engine.cc
Comment (freiling) [inline shell/platform/fuchsia/flutter/engine.cc:261]: nit consider also checking if surface_producer is null

File: lib/ui/compositing/scene_host.cc
Comment (freiling) [inline lib/ui/compositing/scene_host.cc:199]: This should prolly not fail silently, consider at least logging an error if not assert

Comment (freiling) [reply lib/ui/compositing/scene_host.cc:199]: if the weak ptr is null

Comment (freiling) [reply lib/ui/compositing/scene_host.cc:199]: consider assert?

File: shell/platform/fuchsia/flutter/platform_view.cc
Comment (freiling) [inline shell/platform/fuchsia/flutter/platform_view.cc:355]: conisder warning/error/assert

**Review by freiling — APPROVED** [review]
LGTM once testing is finished and comments addressed

**Comment by chinmaygarde** [comment]
@arbreng  The test failure looks related.

**Comment by arbreng** [comment]
> @arbreng The test failure looks related.

Yes, I have some cleanup yet to do on this CL before landing it.  It will be this evening or tomorrow

File: shell/platform/fuchsia/flutter/platform_view.h
Comment (arbreng) [reply shell/platform/fuchsia/flutter/platform_view.h:207]: Good call

File: shell/platform/fuchsia/flutter/platform_view.h
Comment (arbreng) [reply shell/platform/fuchsia/flutter/platform_view.h:75]: I agree w/ this nit but gonna do it later in a formatting CL

File: shell/platform/fuchsia/flutter/platform_view.cc
Comment (arbreng) [reply shell/platform/fuchsia/flutter/platform_view.cc:350]: Yes good point.  It should be guaranteed at this point, so we can check on that

File: shell/platform/fuchsia/flutter/platform_view.cc
Comment (arbreng) [reply shell/platform/fuchsia/flutter/platform_view.cc:362]: Done

File: shell/platform/fuchsia/flutter/platform_view.cc
Comment (arbreng) [reply shell/platform/fuchsia/flutter/platform_view.cc:468]: Unfortunately I cannot change those without doing a soft-transition in the fuchsia SDK code :(

File: shell/platform/fuchsia/flutter/platform_view.cc
Comment (arbreng) [reply shell/platform/fuchsia/flutter/platform_view.cc:946]: Its not unfortunately, they are on C++14

File: shell/platform/fuchsia/flutter/platform_view.cc
Comment (arbreng) [reply shell/platform/fuchsia/flutter/platform_view.cc:415]: Good call: https://rapidjson.org/md_doc_stream.html

File: flow/view_holder.cc
Comment (arbreng) [reply flow/view_holder.cc:123]: Oops, this was the source of a bug I struggling with during testing :)

Good catch :+1:

File: flow/view_holder.cc
Comment (arbreng) [reply flow/view_holder.cc:80]: Ooops!  also good catch :+1:

File: shell/platform/fuchsia/BUILD.gn
Comment (arbreng) [reply shell/platform/fuchsia/BUILD.gn]: Woops, forgot to re-enable

File: shell/platform/fuchsia/flutter/engine.cc
Comment (arbreng) [reply shell/platform/fuchsia/flutter/engine.cc]: Yes I think so, fml::TaskRunner provides such things

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.h
Comment (arbreng) [reply shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.h]: Obsolete

File: shell/platform/fuchsia/flutter/platform_view.h
Comment (arbreng) [reply shell/platform/fuchsia/flutter/platform_view.h]: Obsolete

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc
Comment (arbreng) [reply shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc]: Nothing guarantees this ATM

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc
Comment (arbreng) [reply shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc]: Nice I think that would, I am just being old-fashioned

File: lib/ui/compositing/scene_host.cc
Comment (arbreng) [reply lib/ui/compositing/scene_host.cc]: I mean yes, but it's years old at this point

File: lib/ui/compositing/scene_host.cc
Comment (arbreng) [reply lib/ui/compositing/scene_host.cc:223]: view_holder_id_ == 0 is a special state meaning it hasn't been emplaced in the map yet, so I'd prefer to keep this explicit

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc
Comment (arbreng) [reply shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc]: Obsolete, I'm not changing this part of the code anymore

File: shell/platform/fuchsia/flutter/platform_view.cc
Comment (arbreng) [reply shell/platform/fuchsia/flutter/platform_view.cc]: I noted it in the documentation for CreateView and DestroyView

File: lib/ui/compositing/scene_host.cc
Comment (arbreng) [reply lib/ui/compositing/scene_host.cc:197]: Done

**Pull request merged**
